modules/serverlist.js

```diff
@@ -0,0 +1,81 @@
+// Made by epsilon#3892
+// https://forum.thd.vg/members/16800/
+
+var config = require("../config.json");
+var request = require("request")
+
+module.exports = function(client) {
+	var module = {};
+
+	var channel, message;
+	channel = client.channels.get(config.server_list.channel);
+	if (!channel) return;
+	channel.fetchMessage(config.server_list.message).then(msg => {
+		message = msg;
+		loop();
+	}).catch(err => {
+		return; // Don't do anything if the message doesn't exist
+	});
+
+	function loop() {
+		request({
+		    url: 'https://api.kag2d.com/v1/game/thd/kag/servers?filters=[{"field":"current","op":"eq","value":"true"},{"field":"connectable","op":"eq","value":true},{"field":"currentPlayers","op":"gt","value":"0"}]',
+		    json: true
+		}, function (error, response, body) {
+			if (error) return;
+
+			let servers = body.serverList;
+
+			// Sort servers and players
+			servers = servers.sort((a, b) => {
+				if (a.currentPlayers === b.currentPlayers) {
+					return a.name.toUpperCase().localeCompare(b.name.toUpperCase());
+				}
+				return b.currentPlayers - a.currentPlayers;
+			});
+			servers.forEach(server => {
+				server.playerList.sort((a, b) => {
+					return a.toUpperCase().localeCompare(b.toUpperCase());
+				});
+			});
+
+			// Make message content
+			let players = servers.reduce((t, x) => t + x.currentPlayers, 0);
+			let text = '```md\n' + `# Server list - ${servers.length} ${plural(servers.length, 'server')}, ${players} ${plural(players, 'player')}` + '``````diff\n';
+			text += servers.map(server => {
+				let prefix = (/(?=^KAG Official \w+ (AUS|AU|EU|US|USA)\b)|(?=^Official Modded Server (AUS|AU|EU|US|USA)\b)/g.test(server.name)) ? '-' : '+';
+				let full = (server.playerPercentage >= 1) ? ' [FULL]' : '';
+				let specs = (server.spectatorPlayers > 0) ? ` (${server.spectatorPlayers} spec)` : '';
+				return `${prefix} ${alignText(server.name, 50, -1)} ${alignText(server.currentPlayers, 3, 1)}/${server.maxPlayers}${full}${specs}\n​${server.playerList.join('  ')}`;
+			}).join('\n\n') + '\n```';
+
+			// Update message, channel and presence
+			channel.setName(`${servers.length}-${plural(servers.length, 'server')}_${players}-${plural(players, 'player')}`);
+			client.user.setPresence({ status: 'online', game: { name: `with ${players} ${plural(players, 'fishy', 'ies', 1)}` } });
```

Comment:
so this replaces the playercount module? please remove it and exclude it from the config and index.js.

Reply:
would making it an option in config be sufficient? it uses the server data which would be hard to access from a different module.

Reply:
it is meant to completely replace the playercount module since thats quire inaccurate tbh